interfaces/repo.go

```diff
@@ -764,6 +764,47 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]*ConnRef, err
 	return conns, nil
 }
 
+// ConnectionsForDeviceKey returns all hotplug connections for given device key and interface name.
+func (r *Repository) ConnectionsForDeviceKey(deviceKey, ifaceName string) ([]*ConnRef, error) {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	snapName, err := r.guessSystemSnapName()
+	if err != nil {
+		return nil, err
+	}
+	var conns []*ConnRef
+	for _, slotInfo := range r.slots[snapName] {
+		if slotInfo.HotplugDeviceKey == deviceKey && slotInfo.Interface == ifaceName {
```

Comment:
This ends up as _slotInfo.HotplugKey == hotplugKey_, which is quite nice.

The order of comparisons should be inverted though, for the same reason mentioned above.